checkipaconsistency/freeipaserver.py

```diff
@@ -323,11 +325,60 @@ def _count_certificates(self):
         self._log.debug(r)
         return r
 
+    def _is_conflict_hidden_version(self):
+
+        # vendor version has this format:
+        # 389-Directory/<rpm version> B<build date>
+        def vendorversion_to_rpm_version(vendor):
+            return vendor.split('/')[1].split(' ')[0]
+
+        # return rpm version v1 greater or equal v2
+        def rpm_version_ge_than(version1, version2):
+            res = False
+            list1 = version1.split('.')
+            list2 = version2.split('.')
+            if len(list1) != 4 or len(list2) != 4:
+                res = False
+            else:
+                value1 = 0
+                value2 = 0
+                for x in range(4):
+                    value1 += int(list1[x])*(100**(3-x))
+                    value2 += int(list2[x])*(100**(3-x))
+                if value1 >= value2:
+                    res = True
+            return res
+
+        result = False
+        results = self._search('', '(objectClass=*)', ['vendorVersion'], ldap.SCOPE_BASE)
+        if results and type(results) is list:
+            if len(results) > 0:
+                dn, attrs = results[0]
+                value = vendorversion_to_rpm_version(attrs['vendorVersion'][0])
+                if value is not None:
+                    result = rpm_version_ge_than(value, VERSION_CONFLICT_HIDDEN)
+
+        return result
+
+    # the conflicts filter depends on 389-ds-base version.
+    # They have been hidden from VERSION_CONFLICT_HIDDEN and on
+    def _get_conflicts_filter(self):
+        self._log.info('Getting conflicts filter ...')
+        confl_filter_tmp = '(nsds5ReplConflict=*)'
+        if self._is_conflict_hidden_version():
+            confl_filter = '(&%s(objectclass=ldapsubentry))' % confl_filter_tmp
```

Comment:
Couldn't you do this with: (|(nsDs5Replconflict=*)(&(oc=ldapsubentry)(nsds5replconflict=*))) ? That would work on both cases perhaps ....